mxcubecore/HardwareObjects/BeamlineActions.py

```diff
@@ -185,25 +186,30 @@ def _get_command_object_class(self, path_str):
         return _cls
 
     def init(self):
```

Comment:
```suggestion
    def init(self): # noqa: C901
```
Ruff complains that this function is too complicated ( I guess it counts number of control statements or something, I am not sure). 

You can either silence it (as in suggestion) or split code in different methods, whatever you think would be better.
I imagine silencing this rule is good enough, since it follows the same kind of flow (if -> try/except/else, elif->try/except/else and so on) and is not really confusing.